core/src/main/java/org/apache/iceberg/FileCleanupStrategy.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.function.Consumer;
+import org.apache.iceberg.avro.Avro;
+import org.apache.iceberg.exceptions.NotFoundException;
+import org.apache.iceberg.io.CloseableIterable;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.util.Tasks;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@SuppressWarnings("checkstyle:VisibilityModifier")
+abstract class FileCleanupStrategy {
+  private static final Logger LOG = LoggerFactory.getLogger(FileCleanupStrategy.class);
+
+  protected final FileIO fileIO;
+  protected final ExecutorService planExecutorService;
```

Comment:
@rdblue The planning threadpool and FileIO instance are still accessed in the implementations so that's why they are protected. This is expected because the logic for planning which data files to delete are different in each of the implementations, but we still want the same threadpool across implementations. I see a way to abstract the planning logic but seems less readable.

Reply:
But the other fields can be made private after refactoring, so updated for that.